numbergen/__init__.py

```diff
@@ -208,7 +208,7 @@ def _rational(self, val):
         elif hasattr(val, 'numer'):
             (numer, denom) = (int(val.numer()), int(val.denom()))
         else:
-            param.main.warning("Casting type '%s' to Fraction.fraction"
+            param.main.param.warning("Casting type '%s' to Fraction.fraction"
```

Comment:
I have mixed feelings about this as I am not sure why logging is related to the core of what param offers. I feel many projects will just use the python logging module their own way so I think I would vote against putting them on the parameterized object and maybe consider deprecating these logging features entirely someday.

Reply:
Perhaps instead of being methods, the messaging could be extracted into a separate function, taking the Parameterized as an argument?  That way the existence of this functionality wouldn't affect anything not using it, but it would still be there as a uniform way to use Parameterized objects together with the logging module.

In any case, if the methods will end up being deprecated, then they presumably should *not* move onto the `.param` object; there is little point in moving things and *then* deleting them.  Deprecated things should presumably stay in Parameterized and get deprecated there.

Note that param itself uses `.warning()`, but at least one of those and possibly both are candidates for being exceptions rather than warnings.

Reply:
I think this discussion can continue in #233 and doesn't need to be decided just yet.